experimental/LieAlgebras/src/LieAlgebra.jl

```diff
@@ -216,34 +217,15 @@ end
 #
 ###############################################################################
 
-function general_linear_lie_algebra(R::Ring, n::Int)
-  basis = [(b = zero_matrix(R, n, n); b[i, j] = 1; b) for i in 1:n for j in 1:n]
-  s = ["x_$(i)_$(j)" for i in 1:n for j in 1:n]
-  L = lie_algebra(R, n, basis, s)
-  set_attribute!(L, :type => :general_linear)
-  return L
-end
-
-function special_linear_lie_algebra(R::Ring, n::Int)
-  basis_e = [(b = zero_matrix(R, n, n); b[i, j] = 1; b) for i in 1:n for j in (i + 1):n]
-  basis_f = [(b = zero_matrix(R, n, n); b[j, i] = 1; b) for i in 1:n for j in (i + 1):n]
-  basis_h = [
-    (b = zero_matrix(R, n, n); b[i, i] = 1; b[i + 1, i + 1] = -1; b) for i in 1:(n - 1)
-  ]
-  s_e = ["e_$(i)_$(j)" for i in 1:n for j in (i + 1):n]
-  s_f = ["f_$(i)_$(j)" for i in 1:n for j in (i + 1):n]
-  s_h = ["h_$(i)" for i in 1:(n - 1)]
-  L = lie_algebra(R, n, [basis_e; basis_f; basis_h], [s_e; s_f; s_h])
-  set_attribute!(L, :type => :special_linear)
-  return L
-end
-
-function special_orthogonal_lie_algebra(R::Ring, n::Int)
-  basis = [
-    (b = zero_matrix(R, n, n); b[i, j] = 1; b[j, i] = -1; b) for i in 1:n for j in (i + 1):n
-  ]
-  s = ["x_$(i)_$(j)" for i in 1:n for j in (i + 1):n]
-  L = lie_algebra(R, n, basis, s)
-  set_attribute!(L, :type => :special_orthogonal)
-  return L
+function lie_algebra(
+  gapL::GAP.GapObj,
+  s::Vector{<:VarName}=[Symbol("x_$i") for i in 1:GAPWrap.Dimension(gapL)];
+  cached::Bool=true,
+)
+  @req GAPWrap.IsLieAlgebra(gapL) "gapL must be a Lie algebra."
+  if GAPWrap.IsLieObjectCollection(gapL)
+    return codomain(_iso_gap_oscar_linear_lie_algebra(gapL, s; cached))
+  else
```

Comment:
Perhaps check whether `gapL` is finite dimensional; in principle, one can construct free Lie algebras and finitely presented Lie algebras in GAP, which are (currently) not supported here.